testing/cli/test_cli_sct_compute_mtr.py

```diff
@@ -1,14 +1,78 @@
 import pytest
 import logging
+import numpy
+import nibabel as nib
 
+from spinalcordtoolbox.image import Image
 from spinalcordtoolbox.scripts import sct_compute_mtr
 
 logger = logging.getLogger(__name__)
 
 
 @pytest.mark.sct_testing
 @pytest.mark.usefixtures("run_in_sct_testing_data_dir")
-def test_sct_compute_mtr_no_checks():
-    """Run the CLI script without checking results.
-    TODO: Check the results. (This test replaces the 'sct_testing' test, which did not implement any checks.)"""
-    sct_compute_mtr.main(argv=['-mt0', 'mt/mt0.nii.gz', '-mt1', 'mt/mt1.nii.gz'])
+def test_sct_compute_mtr_results_are_identical(tmp_path):
+    """
+    Test the CLI script to ensure that MTR computed with sct_testing_data
+    images remains the same.
+    """
+
+    mt0_path = 'mt/mt0_reg_slicereg_goldstandard.nii.gz'
+    mt1_path = 'mt/mt1.nii.gz'
+    mtr_groundtruth_path = 'mt/mtr.nii.gz'
+    mtr_output_path = str(tmp_path / 'mtr_output.nii.gz')
+
+    # We increase the threshold here because the default value is 100, but the
+    # original MTR file wasn't clipped to +/- 100.
+    sct_compute_mtr.main(argv=['-mt0', mt0_path, '-mt1', mt1_path,
+                               '-o', mtr_output_path, '-thr', '2000'])
+
+    # Compare ground truth mtr file to new generated one
+    ground_truth_mtr = Image(mtr_groundtruth_path)
+    output_mtr = Image(mtr_output_path)
+
+    # NB: numpy.isfinite is used to exclude nan/inf elements from comparison
+    assert numpy.array_equal(ground_truth_mtr.data[numpy.isfinite(ground_truth_mtr.data)],
+                             output_mtr.data[numpy.isfinite(output_mtr.data)])
```

Comment:
Good news! I was able to fix up the other test, too, so that we can compare the files in `sct_testing_data` _exactly_. 

This confirms that `mtr.nii.gz` was indeed created from `mt0_reg_slicereg_goldstandard.nii.gz` and `mt1.nii.gz` 6 years ago. :P